include/tenstorrent/tt_boot_fs.h

```diff
@@ -57,21 +57,9 @@ typedef struct {
 	uint32_t fd_crc;
 } tt_boot_fs_fd;
 
-typedef int (*tt_boot_fs_read)(uint32_t addr, uint32_t size, uint8_t *dst);
-typedef int (*tt_boot_fs_write)(uint32_t addr, uint32_t size, const uint8_t *src);
-typedef int (*tt_boot_fs_erase)(uint32_t addr, uint32_t size);
-
-typedef struct {
-	tt_boot_fs_read hal_spi_read_f;
-	tt_boot_fs_write hal_spi_write_f;
-	tt_boot_fs_erase hal_spi_erase_f;
-} tt_boot_fs;
-
 typedef struct {
 	uintptr_t magic; /* TT_BOOT_FS_NG_MAGIC */
```

Comment:
As long as the boot fs is initialized earlier than its callers, we can probably just pass around the `const struct device *dev` that represents the flash device without needing to pass around the `tt_boot_fs_ng` pointer or checking for magic.

So likely, we can get rid of all references to `magic` or `TT_BOOT_FS_NG_MAGIC`.